datalad_service/middleware/auth.py

```diff
@@ -0,0 +1,16 @@
+import jwt
+import os
+
+class AuthenticateMiddleware(object):
+    def process_request(self, req, resp):
+        """Process the request before routing it.
+
+        Args:
+            req: Request object that will eventually be
+                routed to an on_* responder method.
+            resp: Response object that will be routed to
+                the on_* responder.
+        """
+        cookies = req.cookies
+        if 'accessToken' in cookies:
+            req.context['user'] = jwt.decode(cookies['accessToken'], key=os.environ['JWT_SECRET'])
```

Comment:
What happens here if the token is invalid?

Reply:
good point. the name, email commit info is set only if the req.context['user'] exists and contains 'name' and 'email' keys. i've wrapped the jwt.decode in general error handling that sets the 'user' field to None.